### PR TITLE
sensor/stmemsc: Align stmemsc i/f to v2.02

### PR DESCRIPTION
**Pull request by avisconti**
Align stmemsc HAL i/f to v2.02

### CODE REVIEW

**Comment by avisconti** [comment]
@stephanosio I added in this PR as a reviewer as I noticed you committed cccbc24c1 few months back (sorry, I missed it). BTW, I would like to review it with you as I want to incorporate the fix (not this one) in the STdC driver repo, so that porting here would be easier (https://github.com/STMicroelectronics/lsm6dso-pid). Any idea?

File: sensor/stmemsc/README [hunk 1]
```diff
@@ -25,18 +25,26 @@ Description:
...
    Define in your 'xyz' driver code the read and write functions that use the
-   I2C or SPI platform driver like the following:
+   I2C or SPI platform driver (plus the optional mdelay function) like the following:
 
    /** Please note that is MANDATORY: return 0 -> no Error.**/
    int platform_wr(void *handle, u8_t reg, const u8_t *bufp, u16_t len);
    int platform_rd(void *handle, u8_t reg, u8_t *bufp, u16_t len);
 
+   /** Component optional fields **/
+   void platform_mdelay(u32_t millisec);
```
Comment (erwango) [inline sensor/stmemsc/README:39]: I guess every user interested in that functionality would need the same zephyr porting function of this.
Can we think of a way to provide it in Zephyr directly, so people don't have to implement it on their side?

File: sensor/stmemsc/_resources/FIFO_Utility_Tool/main.c
Comment (erwango) [inline sensor/stmemsc/_resources/FIFO_Utility_Tool/main.c:1]: Question on this tool: Is that useful within Zephyr context ?

File: sensor/stmemsc/README (hunk 1 above)
Comment (avisconti) [reply sensor/stmemsc/README:39]: > I guess every user interested in that functionality would need the same zephyr porting function of this. Can we think of a way to provide it in Zephyr directly, so people don't have to implement it on their side?

Good idea. 
What I can do is to add stmemsc_delay() in file https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/stmemsc/stmemsc.h:

```
void platform_mdelay(uint32_t ms)
{
  k_msleep(ms);
}
```

 which can then be used in the stmemc based drivers (like here https://github.com/zephyrproject-rtos/zephyr/blob/1f57517e2c5fc923686883781838c3f6fdbec978/drivers/sensor/lsm6dso/lsm6dso.c#L955)

What do you think?

File: sensor/stmemsc/_resources/FIFO_Utility_Tool/main.c
Comment (avisconti) [reply sensor/stmemsc/_resources/FIFO_Utility_Tool/main.c:1]: > Question on this tool: Is that useful within Zephyr context ?

Mmmmh, I'd say no. But it was there also previously. I can remove it if it is a better choice.

File: sensor/stmemsc/README (hunk 1 above)
Comment (erwango) [reply sensor/stmemsc/README:39]: That would be great, but only thing is that can't do it that way. Porting on zephyr symbols should be done in zephyr main repo.
See https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/st_stm32/common/stm32cube_hal.c

File: sensor/stmemsc/_resources/FIFO_Utility_Tool/main.c
Comment (erwango) [reply sensor/stmemsc/_resources/FIFO_Utility_Tool/main.c:1]: Ok,if it was there, it can be kept I guess

File: sensor/stmemsc/README (hunk 1 above)
Comment (avisconti) [reply sensor/stmemsc/README:39]: > That would be great, but only thing is that can't do it that way. Porting on zephyr symbols should be done in zephyr main repo. See https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/st_stm32/common/stm32cube_hal.c

Sure, I meant that I'd do it in zephyr main repo. Seems very similar to HAL_Delay() implementation infact.

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**